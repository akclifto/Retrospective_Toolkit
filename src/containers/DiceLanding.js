/**
 * This class will be used as a container component that holds the other components in the landing page
 */

//Imports
import React from 'react';
import Die from '../components/Dice/Die';
import {sides as sidesConst, themes as themeConst } from '../constants/DieConstants';
import { makeStyles } from '@material-ui/core/styles';
import InfoCard from '../components/InfoCard';
import Grid from '@material-ui/core/Grid'


const useStyles = makeStyles((theme) => ({
  root: {
    flexGrow: 1,
    padding: '5px',
  },
}));

//Directs the browser to statefarm for more information in new tab.
const redirect = (e) => {
  e.preventDefault();
  const url = "http://statefarm.com";
  window.open(url, '_blank');
}


//Returns a landing page for the Dice Game
const DiceLanding = () => {
  const classes = useStyles();

  return (
    
    <div>

      <Grid className={classes.root} 
        container direction="row" justify="center" alignItems="center">
          <Grid item xs={5}>
            <InfoCard
              title="Cube Game"
              body="This cube game is intended to promote communication and understanding within a team."
              body2="How To Play:"
              body3="First, choose an action dice, then roll it. It will roll on a side with a picture. 
              Describe how the picture relates to an experience in software development you have had."
              clicked={redirect}
              />
          </Grid>
        {/** Creates a Die object, contains variable properties*/}
        <Grid item xs={5}>
        <div className={classes.dice}>
        <Die 
          numSides={sidesConst.SIX.sides}
          title={themeConst.Action}
        />
        </div>
      </Grid>
    </Grid>
  </div>
  );
}

export default DiceLanding;
