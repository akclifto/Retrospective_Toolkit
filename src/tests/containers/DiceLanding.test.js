import React from "react";
import { shallow } from "enzyme";
import DiceLanding from "../../containers/DiceLanding";

describe("Container/DiceLanding Testing", () => {
  // Snapshot for LandingPage
  test("SNAPSHOT: diceLanding", () => {
    const wrapper = shallow(<DiceLanding />);
    expect(wrapper).toMatchSnapshot();
  });
});