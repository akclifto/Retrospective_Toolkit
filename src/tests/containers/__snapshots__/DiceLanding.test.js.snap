// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container/DiceLanding Testing SNAPSHOT: diceLanding 1`] = `
<div>
  <WithStyles(ForwardRef(Grid))
    className="makeStyles-root-1"
    container={true}
    direction="row"
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={12}
    >
      <InfoCard
        body="This cube game is intended to promote communication and understanding within a team."
        body2="How To Play: Click 'Start Game' to roll the dice!"
        body3="Each die will land on a random picture.  Describe how the picture relates to an experience you had in software development. To re-roll all die, click the 'Roll It!' button, to re-roll one die, click on a single die to re-roll."
        buttonName="Video Demo"
        buttonOnClick={[Function]}
        title="Cube Game"
      />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={12}
    >
      <ThreeDice />
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</div>
`;
